partiql-lang/src/main/kotlin/org/partiql/lang/eval/Bindings.kt

```diff
@@ -56,6 +56,63 @@ fun PartiqlAst.CaseSensitivity.toBindingCase(): BindingCase = when (this) {
     is PartiqlAst.CaseSensitivity.CaseSensitive -> BindingCase.SENSITIVE
 }
 
+/**
+ * Represents a namespaced global binding.
+ */
+class BindingId(
+    private val qualifier: List<BindingName>,
+    private val id: BindingName
+) : Iterable<BindingName> {
+    /**
+     * For input such as `SELECT a FROM catalog1."schema1".table1`, the identifier is `table1`.
+     */
+    fun getIdentifier(): BindingName = id
+
+    /**
+     * For input such as `SELECT a FROM catalog1."schema1".table1`, the qualifier is `catalog1."schema1"`
+     */
+    fun getQualifier(): List<BindingName> = qualifier
+
+    /**
+     * Returns whether the id is qualified.
+     */
+    fun hasQualifier(): Boolean = qualifier.isNotEmpty()
+
+    /**
+     * Returns a collection of the parts of the identifier.
+     */
+    fun getParts(): List<BindingName> {
+        return qualifier + id
+    }
```

Comment:
If you just have a single list of `BindingName` as I suggested above, the first three functions above aren't needed.

Also, why not use kotlin's read-only property syntax here?

```Kotlin
val parts get() = qualifier + id
```

It's quite identical to your `getParts`, Java even should still see a `getParts` , AFAIK.